### PR TITLE
merge export-parallel

### PR DESCRIPTION
**Pull request by SergeyGaluzo**

### CODE REVIEW

**Review by LTA-Thinking — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                var till = record.Till.ToDateTimeOffset();
+                var globalEndId = _searchService.GetSurrogateId(till.DateTime) - 1; // -1 is so _till value can be used as _since in the next time based export
+
+                var enqueued = groupJobs.Where(_ => _.Id != jobInfo.Id) // exclude coord
+                                        .Select(_ => JsonConvert.DeserializeObject<ExportJobRecord>(_.Definition))
+                                        .Where(_ => _.EndSurrogateId != null) // This is to handle current mock tests. It is not needed but does not hurt.
+                                        .GroupBy(_ => _.ResourceType)
+                                        .ToDictionary(_ => _.Key, _ => _.Max(r => long.Parse(r.EndSurrogateId)));
+
+                await Parallel.ForEachAsync(resourceTypes, new ParallelOptions { MaxDegreeOfParallelism = 4, CancellationToken = cancellationToken }, async (type, cancel) =>
```
Comment (PTaladay) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs:74]: Is there a reason why MaxDegreeOfParallelism is hardcoded to 4? Would this be something that we want to have configurable in the future?

Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs:74]: Unlike number of parallel tasks this is not a strong parameter. I do not expect any differences if value is increased.  So as long as it does not crash for low end systems at 4, we should never have need to change it.

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            _queueClient = queueClient;
+            _searchService = searchService;
+            _logger = loggerFactory.CreateLogger<ExportOrchestratorJob>();
+        }
+
+        internal int NumberOfSurrogateIdRanges { get; set; } = DefaultNumberOfSurrogateIdRanges;
+
+        public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progress, CancellationToken cancellationToken)
+        {
+            var record = JsonConvert.DeserializeObject<ExportJobRecord>(jobInfo.Definition);
```
Comment (PTaladay) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs:48]: Missing EnsureArg.IsNotNull(jobInfo, nameof(jobInfo)); and EnsureArg.IsNotNull(progress, nameof(progress)); ?

File: src/Microsoft.Health.Fhir.CosmosDb.UnitTests/Features/Operations/Reindex/ReindexJobCosmosThrottleControllerTests.cs [hunk 3]
```diff
@@ -46,7 +46,7 @@ public ReindexJobCosmosThrottleControllerTests(ITestOutputHelper output)
             _fhirRequestContextAccessor.RequestContext = fhirRequestContext;
         }
 
-        [Fact]
+        [Fact(Skip = "TODO: Investigate intermittent failures")]
```
Comment (PTaladay) [inline src/Microsoft.Health.Fhir.CosmosDb.UnitTests/Features/Operations/Reindex/ReindexJobCosmosThrottleControllerTests.cs:49]: Is there a work item for this so it doesn't get lost?

Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.CosmosDb.UnitTests/Features/Operations/Reindex/ReindexJobCosmosThrottleControllerTests.cs:49]: Yes

**Review by PTaladay — APPROVED** [review]

**Pull request merged**